app/api/dds/v1/auth_providers_api.rb

```diff
@@ -28,6 +57,113 @@ class AuthProvidersAPI < Grape::API
       get '/auth_providers/:id', root: false, serializer: AuthenticationServiceSerializer do
         AuthenticationService.find(params[:id])
       end
+
+      desc 'List Auth Provider Affiliates' do
+        detail 'List Auth Provider Affiliates'
+        named 'list authentication provider affiliates'
+        failure [
+          [200, 'Success'],
+          [404, 'AuthenticationService does not exist'],
+          [400, 'AuthenticationService does not support affiliate search'],
+          [503, 'Identity Provider Communication Failure']
+        ]
+      end
+      params do
+        requires :id, type: String, desc: 'AuthenticationProvider UUID'
+        requires :full_name_contains, type: String, desc: 'string contained in name(must be at least 3 characters)'
+      end
+      get '/auth_providers/:id/affiliates', root: 'results', each_serializer: AffiliateSerializer do
+        affiliate_params = declared(params, {include_missing: false}, [:full_name_contains])
+        if UserAuthenticationService.where(
+          authentication_service_id: params[:id],
+          uid: affiliate_params[:uid]
+        ).exists?
+          affiliate_exists_error!
+        else
+          auth_service = AuthenticationService.find(params[:id])
+          if auth_service.identity_provider.nil?
+            unsupported_affiliate_search_error!
+          else
+            auth_service.identity_provider.affiliates(affiliate_params[:full_name_contains])
```

Comment:
https://github.com/ruby-ldap/ruby-net-ldap/blob/e4c46a223a19feda78393a793711353aa1febdcd/lib/net/ldap/connection.rb

I really dont think the ruby net-ldap search interface allows you access to pagination. They paginate the results under the hood, but they repeatedly search until all results are returned before returning to the caller with all the results.  They do allow you to send a maximum size, and time value (this might be useful to prevent timeouts), but these just end up minimizing each call to get the next page's results, while still continuously getting each pages results until they are all returned.

If we want to support pagination, we will have to do it wastefully, e.g. get all results, skip to N, add until M, break. This would satisfy clients requirement if they end up needing pagination, but would not prevent server timeouts of memory explosions.